### PR TITLE
ISSUE-788 Setting to enable Push for firebase

### PR DESCRIPTION
**Pull request by quantranhong1999**
resolve https://github.com/linagora/tmail-backend/issues/788

### CODE REVIEW

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/settings/JmapSettings.scala [hunk 1]
```diff
@@ -40,6 +41,8 @@ object JmapSettings {
   case class JmapSettingsValue(value: String) extends AnyVal
 }
 
+trait JmapSettingEntry
```
Comment (vttranlina) [inline tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/settings/JmapSettings.scala:44]: can we have another name?
it is bit confusing with `JmapSettingsEntry`  https://github.com/linagora/tmail-backend/blob/a39701c757e7c6aeac357a20ef196e5d18912876/tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/JmapSettingsGet.scala#L30C12-L30C29

**Review by vttranlina — APPROVED** [review]

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/settings/JmapSettings.scala (hunk 1 above)
Comment (Arsnael) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/settings/JmapSettings.scala:44]: Or refactoring? Actually after thinking again, I think JmapSettingsEntry here would be more accurate (it's one entry of the settings) than yours...

For yours -> JmapSettings? or JmapSettingsObject? or JmapSettingsSingleton? 

The term `Entry` makes me think of something small, atomic

**Review by Arsnael — COMMENTED** [review]
Read it, nothing to add

File: tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/settings/JmapSettings.scala (hunk 1 above)
Comment (vttranlina) [reply tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/settings/JmapSettings.scala:44]: I'm fine with both

**Review by vttranlina — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Pull request merged**